appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/service/FaultToleranceServiceImpl.java

```diff
@@ -120,14 +110,18 @@
     private ScheduledExecutorService delayExecutorService;
 
     @PostConstruct
-    public void postConstruct() throws NamingException {
-        events.register(this);
-        invocationManager = serviceLocator.getService(InvocationManager.class);
-        requestTracingService = serviceLocator.getService(RequestTracingService.class);
-        config = serviceLocator.getService(FaultToleranceServiceConfiguration.class);
-        InitialContext context = new InitialContext();
-        asyncExecutorService = (ManagedExecutorService) context.lookup(config.getManagedExecutorService());
-        delayExecutorService = (ManagedScheduledExecutorService) context.lookup(config.getManagedScheduledExecutorService());
+    public void postConstruct() {
+        try {
+            events.register(this);
+            invocationManager = serviceLocator.getService(InvocationManager.class);
+            requestTracingService = serviceLocator.getService(RequestTracingService.class);
+            config = serviceLocator.getService(FaultToleranceServiceConfiguration.class);
+            InitialContext context = new InitialContext();
+            asyncExecutorService = (ManagedExecutorService) context.lookup(config.getManagedExecutorService());
+            delayExecutorService = (ManagedScheduledExecutorService) context.lookup(config.getManagedScheduledExecutorService());
+        } catch (NamingException namingException) {
+            throw new RuntimeException("FaultToleranceServiceImpl InitialContext creation failure", namingException);
```

Comment:
I'd provide a more useful message
```suggestion
            throw new RuntimeException("Error initialising Fault Tolerance Service: could not perform lookup for configured managed-executor-service or managed-scheduled-executor-service.", namingException);
```